source/material_model/rheology/elasticity.cc

```diff
@@ -298,61 +354,166 @@ namespace aspect
                                                              update_gradients,
                                                              this->introspection().component_indices.velocities[0]));
 
-            // Initialize the evaluator for the old velocity gradients
+            // Initialize the evaluator for the velocity gradients
             evaluator->reinit(in.current_cell, quadrature_positions);
             evaluator->evaluate(solution_values,
                                 EvaluationFlags::gradients);
 
-            const double dte = elastic_timestep();
-            const double dt = this->get_timestep();
+            // Get the compositional fields from the previous timestep $t$,
+            // but only those that represent the stress tensor.
+            // The 'old_solution' has been updated to the full stress tensor
+            // of time $t$ by the iterator splitting step at the beginning
+            // of the current timestep.
+            std::vector<double> old_solution_values(this->get_fe().dofs_per_cell);
+            in.current_cell->get_dof_values(this->get_old_solution(),
+                                            old_solution_values.begin(),
+                                            old_solution_values.end());
 
-            for (unsigned int i=0; i < in.n_evaluation_points(); ++i)
+            // Only create the evaluator the first time we get here
+            // We assume (and assert in parse_parameters) that the n_independent_components
+            // tensor components are the first fields in the correct order.
```

Comment:
I've kept this comment here, because we start the `evaluator_composition` with the first field (`this->introspection().component_indices.compositional_fields[0]`).